pkg/ingress/model_build_listener_rules.go

```diff
@@ -72,6 +74,19 @@ func (t *defaultModelBuildTask) buildListenerRules(ctx context.Context, lsARN co
 	return nil
 }
 
+// sort paths by length, if lengths equal, the path with Exact pathType has precedence
+func (t *defaultModelBuildTask) sortIngressPaths(paths []networking.HTTPIngressPath) []networking.HTTPIngressPath {
+	sort.Slice(paths, func(i, j int) bool {
+		if len(paths[i].Path) == len(paths[j].Path) {
+			if paths[i].PathType != nil {
+				return *paths[i].PathType == networking.PathTypeExact
```

Comment:
does it work for all cases?

Reply:
I thought about these cases, when 2 paths have the same length, the current algorithm will return below results:
1. path[i] is exact type, path[j] is exact type, other type, or just nil - > path[i] precedes 
2. path[i] is other type, path[j] is exact type, other type or just nil -> path[j] precedes
3. path[i] PathType is nil, path[j] is exact type, other type or just nil -> path[j] precedes

So I think that would be sufficient. Since upstream only indicates the exact type should have higher priority over  others.

Reply:
I thought about these cases, when 2 paths have the same length, the current algorithm will return below results:

1. path[i] is exact type, path[j] is exact type, other type, or just nil - > path[i] precedes
2. path[i] is other type, path[j] is exact type, other type or just nil -> path[j] precedes
3. path[i] PathType is nil, path[j] is exact type, other type or just nil -> path[j] precedes

So I think that would be sufficient. Since upstream only indicates the exact type should have higher priority over others.